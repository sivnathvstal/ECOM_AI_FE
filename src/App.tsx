import React from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import { ThemeProvider, createTheme, CssBaseline } from '@material-ui/core';
import { QueryClient, QueryClientProvider } from 'react-query';
import Header from './components/common/Header';
import ProductListingPage from './pages/ProductListingPage';
import ProductDetailsPage from './pages/ProductDetailsPage';
import CartPage from './pages/CartPage';
import { CartProvider } from './context/CartContext';

// Create a client for React Query
const queryClient = new QueryClient();

// Create a theme
const theme = createTheme({
  palette: {
    primary: {
      main: '#3f51b5',
    },
    secondary: {
      main: '#f50057',
    },
  },
  typography: {
    fontFamily: [
      'Roboto',
      '"Helvetica Neue"',
      'Arial',
      'sans-serif',
    ].join(','),
  },
});

const App: React.FC = () => {
  return (
    <QueryClientProvider client={queryClient}>
      <ThemeProvider theme={theme}>
        <CssBaseline />
        <CartProvider>
          <Router>
            <Header />
            <Switch>
              <Route exact path="/" component={ProductListingPage} />
              <Route path="/product/:id" component={ProductDetailsPage} />
              <Route path="/cart" component={CartPage} />
            </Switch>
          </Router>
        </CartProvider>
      </ThemeProvider>
    </QueryClientProvider>
  );
};

export default App; 