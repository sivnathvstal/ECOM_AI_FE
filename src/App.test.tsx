import React from 'react';
import { render, screen, act } from '@testing-library/react';
import App from './App';

describe('App', () => {
  it('renders correctly', async () => {
    let container;
    await act(async () => {
      const rendered = render(<App />);
      container = rendered.container;
    });
    expect(container).toMatchSnapshot();
  });

  it('renders the header', async () => {
    await act(async () => {
      render(<App />);
    });
    expect(screen.getByText(/E-Commerce Shop/i)).toBeInTheDocument();
  });

  it('renders the product listing page by default', async () => {
    await act(async () => {
      render(<App />);
    });
    expect(screen.getByText(/Product Listing/i)).toBeInTheDocument();
  });
}); 