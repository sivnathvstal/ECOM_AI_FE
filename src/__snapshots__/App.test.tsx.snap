// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <a
        class="MuiTypography-root header-title MuiTypography-h6"
        data-testid="app-title"
        href="/"
      >
        E-Commerce Shop
      </a>
      <a
        aria-disabled="false"
        aria-label="Shopping Cart"
        class="MuiButtonBase-root MuiIconButton-root header-cart-button MuiIconButton-colorInherit"
        data-testid="cart-button"
        href="/cart"
        role="button"
        tabindex="0"
      >
        <span
          class="MuiIconButton-label"
        >
          <span
            class="MuiBadge-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zM1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2H1zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
              />
            </svg>
            <span
              class="MuiBadge-badge header-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorSecondary MuiBadge-invisible"
            >
              0
            </span>
          </span>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </a>
    </div>
  </header>
  <div
    class="MuiContainer-root product-listing-container MuiContainer-maxWidthLg"
    data-testid="product-listing-page"
  >
    <h1
      class="MuiTypography-root product-listing-title MuiTypography-h4"
      data-testid="page-title"
    >
      Products
    </h1>
    <div
      class="MuiBox-root MuiBox-root-1 filter-container"
    >
      <div
        class="MuiBox-root MuiBox-root-2 search-container-wrapper"
      >
        <form
          class="MuiPaper-root search-container MuiPaper-elevation1 MuiPaper-rounded"
          data-testid="search-form"
        >
          <div
            class="MuiInputBase-root search-input"
          >
            <input
              aria-label="search products"
              class="MuiInputBase-input"
              data-testid="search-input"
              placeholder="Search products..."
              type="text"
              value=""
            />
          </div>
          <button
            aria-label="search"
            class="MuiButtonBase-root MuiIconButton-root search-button"
            data-testid="search-button"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </form>
      </div>
      <div
        class="MuiFormControl-root category-filter"
        data-testid="category-filter"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink="false"
          id="category-select-label"
        >
          Category
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline Mui-disabled Mui-disabled MuiInputBase-formControl MuiInput-formControl"
          data-testid="category-select"
        >
          <div
            aria-disabled="true"
            aria-haspopup="listbox"
            aria-labelledby="category-select-label category-select"
            class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input Mui-disabled Mui-disabled Mui-disabled"
            id="category-select"
            role="button"
          >
            <span>
              â€‹
            </span>
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput"
            tabindex="-1"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSelect-icon Mui-disabled"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root MuiBox-root-3 loader-container"
      data-testid="loading-indicator"
    >
      <div
        class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
